test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,2361 @@
+        # and system
+        dirs_in_snapshot = ['server', 'job', 'node', 'comm', 'queue', 'hook',
+                            'scheduler', 'reservation', 'resource',
+                            'datastore', 'pbs', 'core_file_bt', 'system',
+                            self.sched_info[self.SCHED_PRIV][0],
+                            self.server_info[self.SVR_PRIV][0],
+                            self.node_info[self.MOM_PRIV][0],
+                            self.server_info[self.ACCT_LOGS][0],
+                            self.comm_info[self.COMM_LOGS][0],
+                            self.server_info[self.SVR_LOGS][0],
+                            self.sched_info[self.SCHED_LOGS][0],
+                            self.node_info[self.MOM_LOGS][0]]
+        for dir in dirs_in_snapshot:
+            rel_path = os.path.join(self.outdir, dir)
+            os.makedirs(rel_path, 0755)
+
+    def __parse_snapshot_args(self, args=None):
+        """
+        Parse the arguments/options provided to pbs_snapshot
+
+        :param args: A dict of arguments/options accepted by pbs_snapshot.
+        "out_dir" is the only mandatory item needed.
+        :type dict
+        """
+
+        if (args is None or ("out_dir" not in args.keys()) or
+                args["out_dir"] is None):
+            self.logger.error("Internal error")
+            sys.exit(1)
+
+        arg_names = args.keys()
+        self.snapshot_name = ntpath.basename(args["out_dir"])
+        self.outdir = os.path.abspath(args["out_dir"])
+        if "diag" in arg_names and args["diag"] is not None:
+            self.diag = args["diag"]
+        if "hostname" in arg_names and args["hostname"] is not None:
+            self.server_host = args["hostname"]
+        if "num_acct_logs" in arg_names and args["num_acct_logs"] is not None:
+            self.num_acct_logs = int(args["num_acct_logs"])
+        if "num_service_logs" in arg_names and \
+                args["num_service_logs"] is not None:
+            self.num_service_logs = int(args["num_service_logs"])
+        if "additional_hosts" in arg_names and \
+                args["additional_hosts"] is not None:
+            arg_no_spaces = "".join((args["additional_hosts"]).split())
+            self.all_hosts.extend(arg_no_spaces.split(","))
+        if "map_file" in arg_names and args["map_file"] is not None:
+            self.mapfile = args["map_file"]
+        if "anonymize" in arg_names and args["anonymize"] is not None:
+            self.anonymize = args["anonymize"]
+            # Create a PBSAnonymizer object
+            del_attrs = [ATTR_v, ATTR_mailfrom, ATTR_managers, ATTR_owner,
+                         ATTR_operators, ATTR_m, ATTR_M, ATTR_aclgroup,
+                         ATTR_acluser, ATTR_aclResvgroup, ATTR_aclResvuser,
+                         ATTR_name, ATTR_jobdir]
+            obf_attrs = [ATTR_euser, ATTR_egroup, ATTR_project, ATTR_A,
+                         ATTR_g]
```

Comment:
ATTR_u is the User_List.  It’ll have user names in it.  It’s analogous to ATTR_g which is being anonymized.  I’m not sure how I feel about having the ability to anonymize without it being robust.  Should you not submit it until it’s ready?